### DISCUSSION TITLE
Adding File System

### DISCUSSION THREAD

**Discussion by cemalgnlts** [CodinGame/monaco-vscode-api#284] [Q&A] [answered]
Hi,

I am trying to add a file system as in the sample code and this is the error I get:
```
Uncaught (in promise) Error: Default api is not ready yet, do not forget to import 'vscode/localExtensionHost' and wait for services initialization
```

I then include this line:
```diff
+ import "vscode/localExtensionHost";
```

But if I import this line, services are no longer initialized. It doesn't give any error and the library doesn't work.

**Comment by cemalgnlts** [comment]
I will follow up on this issue: https://github.com/CodinGame/monaco-vscode-api/issues/283

**Comment by CGNonofr** [comment, marked as answer]
It'll be fixed soon in #285

in the meantime, you can add this workaround:
```typescript
import { IExtensionService } from 'vscode/vscode/vs/workbench/services/extensions/common/extensions'
import { registerServiceInitializeParticipant } from 'vscode/lifecycle'

registerServiceInitializeParticipant(accessor => accessor.get(IExtensionService))
```

**Reply by cemalgnlts** [reply]
Great, now I can continue.
I am trying to add the file system as in the example you gave but I get this error:
```
Extension 'Null Extension Description' failed to update workspace folders: Error: unsupported
```

https://github.com/cemalgnlts/jside/blob/cemalgnlts/dev/src/libs/configuration/init.ts#L81

I'm trying to use [Origin Private File System](https://developer.mozilla.org/en-US/docs/Web/API/File_System_API/Origin_private_file_system), it has the same API as `showDirectoryPicker`.

**Reply by cemalgnlts** [reply]
I missed this line and the problem was solved after I added this line.
```js
workspaceProvider: {
  trusted: true,
  open: async () => false,
  workspace: { workspaceUri }
}
```

Also in the new version `vscode/localExtensionHost` now works correctly.

Thanks.
